### PR TITLE
Fix data races in kafkatest by using atomic increment

### PR DESCRIPTION
**Pull request by satran**

### CODE REVIEW

File: kafkatest/broker.go [hunk 1]
```diff
@@ -183,7 +184,7 @@ func (p *Producer) Produce(topic string, partition int32, messages ...*proto.Mes
 	if p.ResponseError != nil {
 		return 0, p.ResponseError
 	}
-	off := p.ResponseOffset
+	off := atomic.AddInt64(&p.ResponseOffset, int64(len(messages)))
```
Comment (jeremyhuiskamp) [inline kafkatest/broker.go]: `AddInt64` returns the new value.  I think we need to subtract `len(messages)` from `off` before continuing here.

Comment (jeremyhuiskamp) [inline kafkatest/broker.go]: Oh, actually, an atomic add isn't good enough here.  Two goroutines could update the offset, but then add the produced messages to the channel in a different order, resulting in mixed up offsets going down the channel.  I think we have to wrap the whole thing in a mutex.

Comment (satran) [inline kafkatest/broker.go]: True. Fixed.

File: kafkatest/broker.go [hunk 2]
```diff
@@ -195,7 +208,7 @@ func (p *Producer) Produce(topic string, partition int32, messages ...*proto.Mes
 		Partition: partition,
 		Messages:  messages,
 	}
-	p.ResponseOffset += int64(len(messages))
+	p.responseOffset = off
```
Comment (jeremyhuiskamp) [inline kafkatest/broker.go]: I don't think this line should have changed (except to reflect `responseOffset` becoming private).  This just sets it to the same value it was before.

File: kafkatest/broker.go [hunk 3]
```diff
@@ -161,7 +161,8 @@ type Producer struct {
 
 	// ResponseOffset is offset counter returned and incremented by every
 	// Produce method call. By default set to 1.
-	ResponseOffset int64
+	responseOffset int64
```
Comment (jeremyhuiskamp) [inline kafkatest/broker.go]: Very minor thing: documentation should be updated to reflect the changed name :)

File: kafkatest/broker.go (hunk 2 above)
Comment (jeremyhuiskamp) [inline kafkatest/broker.go]: Makes me wonder if there should be some unit tests :-p

**Pull request merged**